test-data/unit/check-newsemanal.test

```diff
@@ -2756,3 +2756,14 @@ def g() -> None:
     x = f()  # E: Need type annotation for 'x'
     y = x
     reveal_type(y)  # N: Revealed type is '__main__.G[Any]'
+
+[case testNewAnalyzerRedefinedNonlocal]
+import typing
+
+def main():
+    bar = []  # type: typing.List[int]
+
+    def foo():
+        nonlocal bar
+        bar = []  # type: typing.List[int]
```

Comment:
I think this should be an error, because this is essentially the same as:
```python
bar: List[int]
bar: List[int]  # Name 'bar' already defined
```

Moreover:
```python-console
>>> import typing
>>> 
>>> def main():
...     bar: typing.List[int] = []
...     def g():
...         nonlocal bar
...         bar: typing.List[int] = []
  File "<stdin>", line 5
SyntaxError: annotated name 'bar' can't be nonlocal
```

Reply:
It generates an error if the functions are annotated. Updated test case to also cover this.